From d48abfc56232fa3e0390c32c507c714caa78a8b1 Mon Sep 17 00:00:00 2001
From: Winson Chung <winsonc@google.com>
Date: Mon, 11 Jun 2018 23:58:27 +0000
Subject: [PATCH 200/306] Revert "Defer stop for all activities becoming
 invisible"

This reverts commit b824f87c65216e4dfe5e9ab0e63795466836bc0f (for b/80313326)

Reason for revert: Cts issues (b/110032866)

Bug: 80313326
Bug: 110032866

Change-Id: Iaca3a8650d30b475ef4331e98b4b14377072ee4d
(cherry picked from commit f49b106dca1b878319a09a1acf8582c05c360a30)
---
 .../core/java/com/android/server/am/ActivityStack.java     | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/services/core/java/com/android/server/am/ActivityStack.java b/services/core/java/com/android/server/am/ActivityStack.java
index cc7a2308834..aea29ac066a 100644
--- a/services/core/java/com/android/server/am/ActivityStack.java
+++ b/services/core/java/com/android/server/am/ActivityStack.java
@@ -1654,13 +1654,6 @@ class ActivityStack<T extends StackWindowController> extends ConfigurationContai
     void addToStopping(ActivityRecord r, boolean scheduleIdle, boolean idleDelayed) {
         if (!mStackSupervisor.mStoppingActivities.contains(r)) {
             mStackSupervisor.mStoppingActivities.add(r);
-
-            // Some activity is waiting for another activity to become visible before it's being
-            // stopped, which means that we also want to wait with stopping this one to avoid
-            // flickers.
-            if (!mStackSupervisor.mActivitiesWaitingForVisibleActivity.isEmpty()) {
-                mStackSupervisor.mActivitiesWaitingForVisibleActivity.add(r);
-            }
         }
 
         // If we already have a few activities waiting to stop, then give up
-- 
2.17.1

